### DISCUSSION TITLE
Breadcrumb Functionality for Segment

### DISCUSSION THREAD

**Discussion by brannytempy** [twilio-labs/paste#2755] [Ideas] [answered]
Hi Paste!

Segment uses breadcrumbs as a form of navigating peers. In the video below, I am traversing 2 source peers.

https://user-images.githubusercontent.com/1590723/197008627-2604f07b-0e37-4f52-8df4-5800a43cc2bf.mp4

This is an important piece of functionality, as it reduces the clicks needed to traverse multiple items.

The Breacrumb component would meet our use case if we could pair them with an IconButton + Menu to mimic our pattern we have today in production. Was this caught in the gap analysis? If not, could we request this be considered?

**Comment by SiTaggart** [comment, marked as answer]
Hi @brannytempy 

This wasn't caught in the gap analysis because it doesn't seem to appear in Evergreen at all. Unless I can't find it.

That being said though, you can easily achieve what you need to achieve with composition. I've create a Codesandbox for you to demonstrate how. https://codesandbox.io/s/breadcrumbs-with-actions-tj29b7?file=/src/index.tsx

It creates a design like the below screen shot and, supports an action or menu or popover or what ever at each level. 

<img width="306" alt="image" src="https://user-images.githubusercontent.com/368249/197085953-eabaf395-cd62-4aab-b6e5-29c9744cb2c1.png">

You need to do a little layout creation, but that is idea of Paste components. You can extend components with other components, in a mix and match kind of creation.

Hope that helps!

**Reply by brannytempy** [reply]
Thanks @SiTaggart ! This is perfect!
